### PR TITLE
apps: upgrade fluentd & fluentd-elasticsearch

### LINKED ISSUE #1222: Upgrade fluentd to v13.7.0
**What Chart should be upgraded?**

upstream/fluentd [chart](https://github.com/bitnami/charts/blob/master/bitnami/fluentd/Chart.yaml)

**Why should this Chart be upgraded?**

- [ ] We need to perform a security upgrade.
- [ ] We need it for a new feature: *which feature (link to blocked issue)*.
- [x] We need to keep up to prevent future issues.
- [ ] Other: *Write the reason here*

**Acceptance criteria**

- I checked the migration of the new Chart:
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).
- [ ] I tested the functionality provided by the new Chart (e.g., Auth flow, Grafana dashboards, etc.)

### PR DESCRIPTION
**Pull request by lunkan93**
**What this PR does / why we need it**:
 Upgrades fluentd & fluentd-elasticsearch
**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #1222 #1223 

**Special notes for reviewer**:
The version of the `kubernetes_metadata_filter` plugin that we have been reliant on for the dot replacement functionality is not compatible with newer versions of fluentd. I added and tested another plugin to our custom image for this functionality and it seems to be a good replacement. 

Created a new [tag](https://github.com/elastisys/fluentd-elasticsearch/tree/v4.3.9-ck8s1), built and pushed a new custom [image](https://github.com/orgs/elastisys/packages/container/fluentd/100682870?tag=v4.3.9-ck8s1) for fluentd-elasticsearch.

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
  - [x] is completely transparent, will not impact the workload in any way.
  - [ ] requires running a migration script.
  - [ ] will create noticeable cluster degradation.
        E.g. logs or metrics are not being collected or Kubernetes API server
        will not be responding while upgrading.
  - [ ] requires draining and/or replacing nodes.
  - [ ] will change any APIs.
        E.g. removes or changes any CK8S config options or Kubernetes APIs.
  - [ ] will break the cluster.
        I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
  - [ ] I upgraded no Chart.
  - [x] I upgraded a Chart and determined that no migration steps are needed.
  - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: WIP-CHANGELOG.md [hunk 1]
```diff
@@ -24,14 +28,9 @@
...
-- Change Trivy Operator Dashboard to only count image states once per image instead for each namespace and resource
-- Rework thanos networkpolicies with generator chart
-- Rework monitoring networkpolicies with generator chart
-
 ### Updated
 
 - Update Trivy Operator Dashboard to improve the user experience
 - Bump sops to version `3.7.3`.
 - Update Falco rules
+- Upgrade fluentd to chart version `5.8.2`, app version `1.16.1` and fluentd-elasticsearch to chart version `13.10.0`, app version `4.3.9`.
```
Comment (robinAwallace) [inline WIP-CHANGELOG.md:36]:
```suggestion
- Upgrade fluentd to chart version `5.8.2`, app version `1.16.1` 
- Upgrade fluentd-elasticsearch to chart version `13.10.0`, app version `4.3.9`.
```

**Review by robinAwallace — APPROVED** [review]

File: WIP-CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,10 @@
 
 - Moved `.monitoring.rook` and `.rookCeph.enabled` keys under `rookCeph.monitoring` and `rookCeph.gatekeeperPsp`
 - The value for `networkPolicies.global.externalLoadBalancer` and `networkPolicies.global.ingressUsingHostNetwork` is now set by the init script, based on the `.global.ck8sCloudProvider` value
+- Change Trivy Operator Dashboard to only count image states once per image instead for each namespace and resource
+- Rework thanos networkpolicies with generator chart
+- Rework monitoring networkpolicies with generator chart
+- New fluentd-elasticsearch custom image tag `v4.3.9-ck8s1` which uses a new plugin for dot replacement, as this functionality has been removed from the `kubernetes_metadata_filter` plugin.
```
Comment (OlleLarsson) [inline WIP-CHANGELOG.md:22]: Remind me, why do we need to build a custom image?
How do I find what the source for the image?
I can't see any updates to our [fork](https://github.com/elastisys/fluentd-elasticsearch)

Comment (lunkan93) [reply WIP-CHANGELOG.md:22]: From what I was able to gather the main reason now is that the Gemfile for the image [upstream](https://github.com/monotek/fluentd-elasticsearch/blob/master/Gemfile) doesn't include some of the plugins that we use:

`fluent-plugin-record-modifier` 
`fluent-plugin-dedot_filter` - Added by me

I just created a [tag](https://github.com/elastisys/fluentd-elasticsearch/tree/v4.3.9-ck8s1) on our fork where you should see the config used for the image i built, e.g. the [Gemfile](https://github.com/elastisys/fluentd-elasticsearch/blob/v4.3.9-ck8s1/Gemfile).

File: WIP-CHANGELOG.md [hunk 3]
```diff
@@ -24,14 +28,10 @@
...
-- Rework thanos networkpolicies with generator chart
-- Rework monitoring networkpolicies with generator chart
-
 ### Updated
 
 - Update Trivy Operator Dashboard to improve the user experience
 - Bump sops to version `3.7.3`.
 - Update Falco rules
+- Upgrade fluentd to chart version `5.8.2`, app version `1.16.1`.
+- Upgrade fluentd-elasticsearch to chart version `13.10.0`, app version `4.3.9`.
```
Comment (OlleLarsson) [inline WIP-CHANGELOG.md:37]:
```suggestion
- Upgrade chart `fluentd` to version `5.8.2`, app version `1.16.1`.
- Upgrade chart `fluentd-elasticsearch` to version `13.10.0`, app version `4.3.9`.
```

File: WIP-CHANGELOG.md (hunk 2 above)
Comment (OlleLarsson) [inline WIP-CHANGELOG.md:22]:
```suggestion
- New fluentd-elasticsearch custom image tag `v4.3.9-ck8s1` which includes a plugin for dot replacement, as this functionality has been removed from the `kubernetes_metadata_filter` plugin.
```

**Review by OlleLarsson — APPROVED** [review]

**Pull request merged**